### DISCUSSION TITLE
somePythonProgrammer JARVIS will not play custom responses to wakewords

### DISCUSSION THREAD

**Discussion by ai096** [N3RDIUM/zen#14] [Q&A]
TLDR: Custom 0.wav is replaced with default 0.wav voiceline in response to wakeword 'jarvis' 
(automatically replaced back to original when I speak)

I'm building a tech project with your quality JARVIS assistant and am trying to input custom voice lines. 
I'm just testing it from the vanilla state where it responds with "0.wav" through "6.wav" 
(I've replaced these 6 lines with Arnold Schwarzenneger quotes renamed 0 through 6 .wavs) When I change the voice lines that play when the wakeword "JARVIS" is uttered & 0.wav should play, instead it just changes back to the default 0.wav voice. 
Is there a way to change this for custom responses?

File "C:\[edited out]\AppData\Local\Programs\Python\Python39\lib\site-packages\playsound.py", line 64, in winCommand
    raise PlaysoundException(exceptionMessage)
playsound.PlaysoundException:
    Error 259 for command:
        play speech_tts/0.wav wait

references:
C:\[edited out]\somePythonProgrammer - JARVIS\speech\speaker.py
line 21
    def initialize(self):
        self.SAVE_PATH = "speech_tts/"

In reference to:
https://github.com/somePythonProgrammer/JARVIS
![speech_tts](https://user-images.githubusercontent.com/80369652/164149434-01d99a80-3891-4788-80c9-1cc445c64324.png)
![speech_stt](https://user-images.githubusercontent.com/80369652/164149437-88248a29-79ae-48e8-acc2-6e013ce4e202.png)
![JARVIS](https://user-images.githubusercontent.com/80369652/164151647-eceedca6-02db-4121-b037-98f5e7397c05.png)

**Comment by N3RDIUM** [comment]
I think you've misunderstood some fundamentals. The assistant automatically converts text to speech. You have to mention your responses in the training data of the chatbot. If your question is about the wakeword engine, you can find more info [here](https://github.com/Picovoice/porcupine). If your question is about custom responses, go to the `conversation_data/data.py` file, there you will see a really messy 2d array called `ConversationData`. Just go ahead and add another array of conversations in `string` format. If its confusing, just add the following lines of code for a little test:
```
ConversationData.append([
    "hello!", #This is a test
    "Hi!", #Add your conversations here
    #You can add more!
])

ConversationData.append([
    "Are you robotic?", #Another test
    "I can't inhabit a robotic body yet, but I can entertain you.", #More data
    #You can add more!
])
```

**Reply by ai096** [reply]
Well the question was with custom voice lines vocoded from other sources. So there's no way to have it respond with a premade wav for specific queries in conversation_data.py like the Arnold Schwarzenegger quotes or other voice lines converted into wavs? What about training different tacotron2 voice models? Really cool stuff how it responds to wakewords.

**Reply by N3RDIUM** [reply]
You just gave me a good idea! Maybe I'll add that in an update in the near future.
